src/Native/Unix/System.Native/pal_networking.cpp

```diff
@@ -1525,8 +1525,13 @@ extern "C" Error SystemNative_SetIPv6MulticastOption(intptr_t socket, int32_t mu
 
     ipv6_mreq opt;
     memset(&opt, 0, sizeof(ipv6_mreq));
+
+#if IPV6MR_INTERFACE_UNSIGNED
     opt.ipv6mr_interface = static_cast<unsigned int>(option->InterfaceIndex);
-    
+#else
+    opt.ipv6mr_interface = option->InterfaceIndex;
+#endif
```

Comment:
Nit: consider instead:
```C++
opt.ipv6mr_interface =
#if IPV6MR_INTERFACE_UNSIGNED
    static_cast<unsigned int>(option->InterfaceIndex);
#else
    option->InterfaceIndex
#endif
```
That's mostly a matter of style, so if you prefer what you've got, ok; I prefer the suggested one simply because it helps to highlight what's different between the lines, and a reader doesn't have to examine the assignments to determine if the left-hand side of the assignment is different.